independent-projects/resteasy-reactive/server/runtime/src/main/java/org/jboss/resteasy/reactive/server/handlers/ResourceLocatorHandler.java

```diff
@@ -79,15 +79,17 @@ public void onComplete(Throwable throwable) {
         }
 
         RequestMapper<RuntimeResource> mapper = target.get(requestContext.getMethod());
+        RequestMapper.RequestMatch<RuntimeResource> res = findRequestMatch(mapper, requestContext);
         boolean hadNullMethodMapper = false;
         if (mapper == null) {
             mapper = target.get(null); //another layer of resource locators maybe
             // we set this without checking if we matched, but we only use it after
             // we check for a null mapper, so by the time we use it, it must have meant that
             // we had a matcher for a null method
             hadNullMethodMapper = true;
+            res = findRequestMatch(mapper, requestContext);
```

Comment:
Here it seems we're invoking `findRequestMatch()` again discarding the first value of `res`, but probably this only happens when `mapper` was `null` so `res` was also `null`. It makes it a bit hard to read, though.